yt_dlp/postprocessor/movefilesafterdownload.py

```diff
@@ -20,33 +26,77 @@ def __init__(self, downloader=None, downloaded=True):
     def pp_key(cls):
         return 'MoveFiles'
 
+    def move_file_and_write_to_info(self, info_dict, relevant_dict=None, output_file_type=None):
+        relevant_dict = relevant_dict or info_dict
+        if 'filepath' not in relevant_dict:
+            return
+
+        output_file_type = output_file_type or ''
+        current_filepath, final_filepath = self.determine_filepath(info_dict, relevant_dict, output_file_type)
+        move_result = self.move_file(info_dict, current_filepath, final_filepath)
+
+        if move_result:
+            relevant_dict['filepath'] = move_result
+        else:
+            del relevant_dict['filepath']
+
+    def determine_filepath(self, info_dict, relevant_dict, output_file_type):
+        current_filepath = relevant_dict['filepath']
+        prepared_filepath = self._downloader.prepare_filename(info_dict, output_file_type)
+
+        if (output_file_type == 'thumbnail' and info_dict['__multiple_thumbnails']) or output_file_type == 'subtitle':
+            desired_extension = ''.join(Path(current_filepath).suffixes[-2:])
+        else:
+            desired_extension = Path(current_filepath).suffix
+
+        return current_filepath, replace_extension(prepared_filepath, desired_extension)
```

Comment:
let's do this so you don't need to touch `replace_extension` and the merge conflict can be resolved
```suggestion
        return current_filepath, replace_extension(prepared_filepath, desired_extension[1:])
```